### PR TITLE
Issue #26 - Get tasks by status.

### PR DESCRIPTION
**Pull request by mtwn105**
Modified the get tasks API with status query param to fetch tasks by status if provided. 
Added the comments as well the swagger documentation

### CODE REVIEW

**Comment by mtwn105** [comment]
This will close issue #26

File: taskmanager-api/src/main/java/com/cardinity/taskmanager/service/TaskService.java [hunk 1]
```diff
@@ -126,8 +126,15 @@ public Task getTaskById(@NotNull long taskId){
     }
 
 
-    public List<Task> getTasks(){
-        List<Task> tasks = this.taskRepository.findAll();
+    public List<Task> getTasks(TaskStatus status) {
+        List<Task> tasks;
+        if (status != null) {
```
Comment (ahadahmed) [inline taskmanager-api/src/main/java/com/cardinity/taskmanager/service/TaskService.java:131]: This can be done with ExampleMatcher

Comment (mtwn105) [reply taskmanager-api/src/main/java/com/cardinity/taskmanager/service/TaskService.java:135]: Created a commit where used `ExampleMatcher` removed method from the repository as well.

File: taskmanager-api/src/main/java/com/cardinity/taskmanager/service/TaskService.java [hunk 2]
```diff
@@ -130,7 +134,10 @@ public Task getTaskById(@NotNull long taskId){
...
         if (status != null) {
             // Find all tasks by status if status provided in query params
-            tasks = this.taskRepository.findAllByTaskStatus(status);
+            Task task = new Task(null, null);
+            task.setTaskStatus(status);
+            Example<Task> taskStatusExample = Example.of(task, ExampleMatcher.matchingAll().withMatcher("taskStatus", exact()));
+            tasks = this.taskRepository.findAll(taskStatusExample);
         } else {
             // Find all tasks if status not provided in query params
             tasks = this.taskRepository.findAll();
```
Comment (ahadahmed) [inline taskmanager-api/src/main/java/com/cardinity/taskmanager/service/TaskService.java:143]: When you use exmaple matcher, this find() is redundant.

**Comment by ahadahmed** [comment]
Closing this pull request as the issue is already resolvded.

**Pull request closed without merging**